nexus/lib/nexus/peer.ex

```diff
@@ -0,0 +1,275 @@
+    %RTPCodecParameters{
+      payload_type: 96,
+      mime_type: "video/H264",
+      clock_rate: 90_000
+    }
+  ]
+
+  @opts [
+    ice_servers: [%{urls: "stun:stun.l.google.com:19302"}],
+    audio_codecs: @audio_codecs,
+    video_codecs: @video_codecs
+  ]
+
+  @type id :: String.t()
+
+  @spec start_link(term(), term()) :: GenServer.on_start()
+  def start_link(args, opts) do
+    GenServer.start_link(__MODULE__, args, opts)
+  end
+
+  @spec apply_sdp_answer(id(), String.t()) :: :ok
+  def apply_sdp_answer(id, answer_sdp) do
+    GenServer.call(registry_id(id), {:apply_sdp_answer, answer_sdp})
+  end
+
+  @spec add_ice_candidate(id(), String.t()) :: :ok
+  def add_ice_candidate(id, body) do
+    GenServer.call(registry_id(id), {:add_ice_candidate, body})
+  end
+
+  @spec send_pli(id()) :: :ok
+  def send_pli(id) do
+    GenServer.cast(registry_id(id), :send_pli)
+  end
+
+  @spec notify(id(), term()) :: :ok
+  def notify(id, noti) do
+    GenServer.cast(registry_id(id), {:notification, noti})
+  end
+
+  @spec registry_id(id()) :: term()
+  def registry_id(id), do: {:via, Registry, {Nexus.PeerRegistry, id}}
+
+  @impl true
+  def init([id, channel, peer_ids]) do
+    Logger.debug("Starting new peer #{id}")
+    {:ok, pc} = PeerConnection.start_link(@opts ++ [controlling_process: self()])
+    Process.monitor(pc)
+    Logger.debug("Starting peer connection #{inspect(pc)}")
+
+    state = %State{
+      id: id,
+      channel: channel,
+      pc: pc
+    }
+
+    {:ok, state, {:continue, {:initial_offer, peer_ids}}}
+  end
+
+  @impl true
+  def handle_continue({:initial_offer, peer_ids}, %{pc: pc} = state) do
+    Logger.debug("Creating initial SDP offer for #{state.id}")
+
+    outbound_tracks = setup_transceivers(pc, peer_ids)
+
+    state = State.send_offer(state)
+
+    {:noreply, %{state | outbound_tracks: outbound_tracks}}
+  end
+
+  @impl true
+  def handle_call({:apply_sdp_answer, answer_sdp}, _from, state) do
+    state =
+      if State.offer_pending?(state) do
+        state
+        |> State.apply_answer(answer_sdp)
+        |> State.flush_candidates()
+        |> State.send_notifications()
+      else
+        Logger.debug("Ignoring SDP answer for #{state.id}: no pending offer")
+        state
+      end
+
+    {:reply, :ok, state}
+  end
+
+  @impl true
+  def handle_call({:add_ice_candidate, body}, _from, state) do
+    # TODO: this is not implemented as the RFC requires
+    candidate =
+      body
+      |> Jason.decode!()
+      |> ICECandidate.from_json()
+
+    state = State.add_candidate(state, candidate)
+
+    {:reply, :ok, state}
+  end
+
+  @impl true
+  def handle_cast(:send_pli, %{pc: pc} = state) do
+    inbound_video_track_id = state.inbound_tracks.video
+
+    unless is_nil(inbound_video_track_id) do
+      :ok = PeerConnection.send_pli(pc, inbound_video_track_id)
+    end
+
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_cast({:notification, {:forward, peer, obt}}, state) do
```

Comment:
I believe that using PubSub terms would fit here better

Reply:
You mean just changing `:forward` to `:subscribe`, or have the entire information exchange happen through a pubsub?

Reply:
We can try with renamings at first :thinking: